orca_detector/database_parser.py

```diff
@@ -1,95 +1,257 @@
+        Returns a list containing (label, 'file:start:frames') of each segment.
+        
+        Any segments that are too short are dropped.
+    """
+    
     with sf.SoundFile(file) as wav_file:
-        # more than 1 sec
-        if wav_file.frames > wav_file.samplerate:
-            frames = int(params.file_max_size_seconds*wav_file.samplerate)
-            file_parts = np.arange(0, wav_file.frames, frames)
-            return [[label, f"{file}:{int(start)}:{frames}"] for start in file_parts]
+        # make sure sample is long enough
+        min_frames = int(sample_len * wav_file.samplerate)  # e.g. 2 * 16000
+        if wav_file.frames > min_frames:
+            file_parts = np.arange(0, wav_file.frames, min_frames)
+            sample_list = [[label, '{}:{}:{}'.format(file, int(start), min_frames)] for start in file_parts]
+
+            # truncate final sample which will be shorter than min required for a spectrogram
+            del sample_list[-1]
+            return sample_list
         else:
             return []
 
-
 def _quantize_samples(samples):
+    """
+        Quantizes a list of samples (label, file path), returns a flattened list.
+    """
     quantized_samples = [_quantize_sample(label, file) for [
         label, file] in samples]
     flat_quantized_samples = [
         item for sublist in quantized_samples for item in sublist]
     return flat_quantized_samples
 
+def _onehot(labels, data_path=DATA_PATH):
+    """
+        Converts labels from strings to integers.  All classes that aren't in
+        orca_params.CLASSES are mapped to 'Other'.
+        
+        Returns np.array[observations, classes] representing the labels
+    """
+    
+    # convert to lists (zip generates tuples)
+    labels = list(labels)
 
-def _onehot(labels):
-    classes = len(set(labels))
+    # with label encoder, map all undesired classes to "Other"
+    ohe_classes = orca_params.CLASSES + ['Other']
     encoder = LabelEncoder()
-    encoder.fit(list(labels))
+    encoder.fit(ohe_classes)
+    
+    # create a list holding the int class labels
+    for i in range(len(labels)):
+        if not labels[i] in orca_params.CLASSES:
+            labels[i] = 'Other'
     encoded_labels = encoder.transform(labels)
 
-    onehot_encoded_labels = np.zeros((len(encoded_labels), classes))
+    # build into a numpy array to return
+    onehot_encoded_labels = np.zeros((len(encoded_labels), len(ohe_classes)))
     onehot_encoded_labels[np.arange(len(encoded_labels)), encoded_labels] = 1
+    
+    # save LabelEncoder so inverse transforms can be recovered
+    label_encoder_file = os.path.join(data_path, 'label_encoder.p')
+    # rename old files instead of overwriting.
+    if os.path.exists(label_encoder_file):
+        renamed_file = '{}-old'.format(label_encoder_file)
+        os.rename(label_encoder_file, renamed_file)
+        print('WARNING: Renamed previous file to {}'.format(renamed_file))
```

Comment:
ok, that's awesome!  I wasn't aware of it, but am not surprised that somebody decided to implement that.   I'll file as a cleanup item on the issue tracker, as I think it's a matter of aesthetics, rather than functionality.